### PR TITLE
SUBMARINE-335. Add a guide of yarn runtime with docker containers

### PR DESCRIPTION
**Pull request by oliverhu**
### What is this PR for?
Provide an example and instructions for how to run submarine with tony runtime using Docker.

### What type of PR is it?
Documentation

### Todos
* [ ] - Pending access to DockerHub apache repo to push the default docker image.

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-335

### How should this be tested?
* Follow the instructions uploaded and it should work under individual's dev environment

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by liuxunorg** [comment]
@oliverhu 
Please fixed travis failure. 
In addition you need change PR title to `SUBMARINE-335. Add a ....`

File: dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh [hunk 1]
```diff
@@ -33,27 +33,37 @@ else
...
+  WORKER_CMD="myvenv.zip/$WORKER_CMD"
+fi
 
 SUBMARINE_VERSION=0.3.0-SNAPSHOT
-HADOOP_VERSION=2.9
-SUBMARINE_PATH=/opt/submarine-current
-HADOOP_CONF_PATH=/usr/local/hadoop/etc/hadoop
-MNIST_PATH=/home/yarn/submarine
+SUBMARINE_HADOOP_VERSION=2.9
+SUBMARINE_PATH=/home/pi/apache/submarine/submarine-all/target
```
Comment (yuanzac) [inline dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh]: Mini-submarine is docker image which is created by the script of build_mini-submarine.sh. Users can tests submarine job with it. Mini-submarine has a one node hadoop cluster and submarine libs. The script of run_submarine_mnist_tony.sh is supposed to be executed in it for users' convenience. So SUBMARINE_PATH, HADOOP_CONF_PATH, MNIST_PATH are set according to the mini-submarine configuration.
It's better for us not to change their values. Or we need changes the values both in run_submarine_mnist_tony.sh and build_mini-submarine.sh.

File: dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh [hunk 2]
```diff
@@ -33,27 +33,37 @@ else
...
+else
+  WORKER_CMD="myvenv.zip/$WORKER_CMD"
+fi
 
 SUBMARINE_VERSION=0.3.0-SNAPSHOT
-HADOOP_VERSION=2.9
-SUBMARINE_PATH=/opt/submarine-current
-HADOOP_CONF_PATH=/usr/local/hadoop/etc/hadoop
-MNIST_PATH=/home/yarn/submarine
+SUBMARINE_HADOOP_VERSION=2.9
```
Comment (yuanzac) [inline dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh]: HADOOP_VERSION needs to be changed to SUBMARINE_HADOOP_VERSION in the line 407 of InstallationGuide.md as well.

File: dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh [hunk 3]
```diff
@@ -65,4 +75,5 @@ ${JAVA_CMD} -cp ${SUBMARINE_PATH}/submarine-all-${SUBMARINE_VERSION}-hadoop-${HA
  --worker_launch_cmd "${WORKER_CMD}" \
  --ps_launch_cmd "myvenv.zip/venv/bin/python mnist_distributed.py --steps 2 --data_dir /tmp/data --working_dir /tmp/mode" \
  --insecure \
- --conf tony.containers.resources=${MNIST_PATH}/myvenv.zip#archive,${MNIST_PATH}/mnist_distributed.py,${SUBMARINE_PATH}/submarine-all-${SUBMARINE_VERSION}-hadoop-${HADOOP_VERSION}.jar
+ --conf tony.containers.resources=${MNIST_PATH}/myvenv.zip#archive,${MNIST_PATH}/mnist_distributed.py,${SUBMARINE_PATH}/submarine-all-${SUBMARINE_VERSION}-hadoop-${SUBMARINE_HADOOP_VERSION}.jar \
+ $DOCKER_CONF
```
Comment (yuanzac) [inline dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh]: Add a new line at the end of file

File: docs/helper/docker/tensorflow/mnist/Dockerfile.tony.tf.mnist.tf_1.13.1 [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ENV HADOOP_CONF_DIR=/opt/hadoop-$HADOOP_VERSION/etc/hadoop
+ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+
+# Crate user, make sure the user groups are the same as your host
+RUN groupadd -g 5000 hadoop
+RUN useradd -u 1000 -g hadoop pi
+RUN mkdir /home/pi
+RUN chown pi:hadoop /home/pi
+RUN mkdir /tmp/mode
+RUN chmod 777 /tmp/mode
```
Comment (yuanzac) [inline docs/helper/docker/tensorflow/mnist/Dockerfile.tony.tf.mnist.tf_1.13.1]: Add a new line at the end of file

**Comment by oliverhu** [comment]
@liuxunorg added license to fix travis failures & updated PR title (added the missing dot)

File: dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh (hunk 2 above)
Comment (oliverhu) [reply dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh]: @yuanzac updated InstallationGuide

File: dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh (hunk 1 above)
Comment (oliverhu) [reply dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh]: i see, these changes are not committed either. Reverted

Comment (oliverhu) [reply dev-support/mini-submarine/submarine/run_submarine_mnist_tony.sh]: also added some comments to instruct users how to adapt these to local env

**Comment by liuxunorg** [comment]
@oliverhu Thank you contribution.
Will merge if no more comments.

**Pull request closed without merging**